### PR TITLE
Fixed queryForCurrentValue with array values

### PR DESCRIPTION
**Pull request by TheOtherDude**
### This project uses [semver](semver.org), please check the scope of this pr:

- [ ] #none# - documentation fixes and/or test additions
- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

We have been using this branch in our app for months now, but I hesitated to make a PR because I didn't think it was clean enough, but we can just make it cleaner together if you guys have any ideas.

# CHANGELOG

* Fixed queryForCurrentValue with array values

### CODE REVIEW

File: addon/list-utils.js [hunk 1]
```diff
@@ -142,7 +142,12 @@ export function getItemsFromEmberData ({value, modelDef, data, bunsenId, store,
   const modelType = modelDef.modelType || 'resources'
   const {labelAttribute, queryForCurrentValue, valueAttribute} = modelDef
   const valueAsId = get(value, bunsenId)
-  const actuallyFindCurrentValue = keepCurrentValue && queryForCurrentValue && valueAsId !== undefined
+  let arrayValues
+  if (typeOf(valueAsId) === 'array' && queryForCurrentValue) {
```
Comment (sophypal) [inline addon/list-utils.js]: `Array.isArray`

File: addon/list-utils.js [hunk 2]
```diff
@@ -142,7 +142,12 @@ export function getItemsFromEmberData ({value, modelDef, data, bunsenId, store,
...
+    arrayValues = valueAsId.asMutable()
+    arrayValues = RSVP.Promise.all(arrayValues.map((id) => store.findRecord(modelType, id)))
```
Comment (sophypal) [inline addon/list-utils.js]: No need to call `asMutable` if you're not going to mutate it. You might consider just inlining it with the `map`

Comment (sophypal) [inline addon/list-utils.js]: Help me understand the use case. Is this for `multi-select`? Also, needs tests.

**Comment by sophypal** [comment]
Help me understand the use case. Is this for multi-select? Also, needs tests.

**Comment by TheOtherDude** [comment]
Yes, this is for multi-selects. 

Also, note that, unlike with single-select, currently selected records as always included in the drop down. I find this to be a useful feature because you can easily view and uncheck selections, but I wanted to bring it up because I could see others regarding it as a bug.

File: addon/list-utils.js (hunk 2 above)
Comment (TheOtherDude) [reply addon/list-utils.js]: I added a comment explaining why I'm calling `asMutable()`

**Comment by sophypal** [comment]
Thanks @TheOtherDude

**Comment by sophypal** [comment]
👍

**Pull request merged**